cdap-docs/build.sh

```diff
@@ -198,36 +203,52 @@ function build_javadocs() {
   # Used by reference manual to know to copy the javadocs
   USING_JAVADOCS="true"
   export USING_JAVADOCS
+  local errors
   local javadoc_type=${1}
   local title="Building Javadocs: '${javadoc_type}'"
   display_start_title "${title}"
-  local warnings
+  set_version
   check_build_rst
-  set_environment    
+  set_environment
   if [[ ${DEBUG} == ${TRUE} ]]; then
     local debug_flag="-X"
   else
     local debug_flag=''
   fi
   if [[ ${javadoc_type} == ${ALL} ]]; then
-    javadoc_run="javadoc:aggregate"
+    javadoc_run="javadoc:aggregate -P release"
   else
-    javadoc_run="site"
+    javadoc_run="site -P templates"
   fi
   local start=`date`
   cd ${PROJECT_PATH}
   MAVEN_OPTS="-Xmx4g -XX:MaxPermSize=256m" # match other CDAP builds
-  mvn clean package ${javadoc_run} -P examples,templates,release -DskipTests -Dgpg.skip=true -DisOffline=false ${debug_flag}
-  warnings=$?
-  if [[ ${warnings} -eq 0 ]]; then
-    echo
-    echo "Javadocs Build Start: ${start}"
-    echo "                 End: `date`"
+  local temp_repo="-Dmaven.repo.local=/tmp/repo-cdap-docs/${GIT_BRANCH}"
+  echo "Using temp_repo '${temp_repo}'"
+  if [[ -d ${temp_repo} ]]; then
+    rm -rf ${temp_repo}
+  fi
+  echo "========================================================"
+  echo "Building and installing CDAP to ${temp_repo}"
+  mvn clean install ${temp_repo} -P examples,templates,release -DskipTests -Dgpg.skip=true
+  errors=$?
+  if [[ ${errors} -eq 0 ]]; then
+    echo "========================================================"
+    echo "Building and installing CDAP finished; running Javadocs"
+    mvn ${javadoc_run} ${temp_repo} -DskipTests -Dgpg.skip=true -DisOffline=false ${debug_flag}
+    errors=$?
```

Comment:
probably better to not shadow/reassign the same variable outside the conditional... ie, use a different variable here, and add a ``return`` in the error section below